Microsoft.Dynamics365.UIAutomation.Api.UCI/WebClient.cs

```diff
@@ -4205,16 +4205,15 @@ internal BrowserCommandResult<bool> OpenLookupRecord(int index)
         {
             return this.Execute(GetOptions("Select Lookup Record"), driver =>
             {
-                if (driver.HasElement(By.XPath(AppElements.Xpath[AppReference.Lookup.LookupResultRows])))
-                {
-                    var rows = driver.FindElements(By.XPath(AppElements.Xpath[AppReference.Lookup.LookupResultRows]));
+                driver.WaitForTransaction();
 
-                    if (rows.Count > 0)
-                        rows.FirstOrDefault().Click(true);
-                }
-                else
+                var rows = driver.FindElements(By.XPath(AppElements.Xpath[AppReference.Lookup.LookupResultRows]));
+                if (!rows.Any())
+                {
                     throw new NotFoundException("No rows found");
+                }
 
+                rows.ElementAt(index).Click();

```

Comment:
@ewingjm - Could you add a validation check (and error) if for some reason the number of rows is less than the supplied index value?

Reply:
Sorry @TYLEROL - was travelling all day yesterday and didn't get round to this. I'm happy to do another PR for it. I believe `ElementAt(index)` will throw an `IndexOutOfRangeException`  - but we could throw that as an inner exception to a more detailed `IndexOutOfRangeException` that specifically says the lookup record at the given index wasn't in the flyout?